### PR TITLE
feat: creating a footer

### LINKED ISSUE #16: [FEATURE] Add footer
As a user, I want to see a footer containing the logo of the organisation.

**Acceptance criteria (AC)**

- [ ] Under the pages folder, create a folder called components
- [ ] Create a file called Footer.js
- [ ] Create a footer using full with of the device
- [ ] Footer showing on bottom of the page
- [ ] Logo is showing in the footer
- [ ] a copyright section is added below
- [ ] No additional styling needed

**Wireframe**
Footer

**Logo**
logo

### PR DESCRIPTION
**Pull request by CBID2**
_For an example of how to fill this template out, [see this Pull Request in the Wiki](https://github.com/YurisCodingClub/DevCV/wiki/How-to-create-a-pull-request)._

## Description
This PR creates a footer.

## Fixes Issue
Closes #16 

## Acceptance Criteria

- [x] Under the pages folder, create a folder called components
- [x]  Create a file called Footer.js
- [x]  Create a footer using full with of the device
- [x]  Footer showing on bottom of the page
- [x]  Logo is showing in the footer
- [x]  a copyright section is added below
- [x]  No additional styling needed

## Type of Changes

Make sure your code follows the code style of this project.
Make sure your code is up2date.
The title of the PR is the same as the title of the issue.

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|   x  | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Footer(Before)

### CODE REVIEW

**Comment by CBID2** [comment]
> Nice start so far @CBID2 🙂 

Thanks @YuriDevAT! 😊 I just need to figure out a way to add the CSS 🤔

**Comment by CBID2** [comment]
Hi @YuriDevAT! :) My PR is officially ready for review! :)

**Comment by YuriDevAT** [comment]
> > Nice start so far @CBID2 🙂
> 
> Thanks @YuriDevAT! 😊 I just need to figure out a way to add the CSS 🤔

Oh sorry, I missed this comment!

**Comment by YuriDevAT** [comment]
Can you share a screenshot with the footer showing on your screen?

I implemented it (since it wasn't already and there were no further instruction on how to implement it) by adding it to the `index.js` file under the `<main>`.
Code snippet of the index file

But I get a compile error because the logo was not imported right.
Failed to compile

**Review by YuriDevAT — CHANGES_REQUESTED** [review]
Thank you so much for your contribution to the project 😃 I left some comments. After making some changes I am happy to merge your PR. You are doing such a great job 🔥 

Always keep in mind to test your implementation, checking, if everything is working on your side as expected. It makes PR approvals way faster. 🚀

File: pages/components/Footer.js
Comment (CBID2) [reply pages/components/Footer.js:8]: Hi @YuriDevAT. I was struggling with moving it to the bottom `position: relative;` So I found [this tutorial](https://dev.to/gedalyakrycer/ohsnap-this-how-to-pin-footers-in-react-3k1k) to help me.

Comment (YuriDevAT) [reply pages/components/Footer.js:8]: Thanks for sharing. The tutorial is good! Is it working as expected on your side? As mentioned in the tutorial, the position attribute for the footer depends on the settings of the main component. Since you did not make any changes there I guess it will not work.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:3]: Where the images were stored was fine, since we are going to work with storybook anyways. The problem was that the path was not set correctly.

**Comment by CBID2** [comment]
Hi @YuriDevAT! :) I revised the file path for the image import component, but for some reason, it does not show up in the deployment. Did I do something wrong? Also, I use `height ="auto"` like you suggested?

**Comment by YuriDevAT** [comment]
> Hi @YuriDevAT! :) I revised the file path for the image import component, but for some reason, it does not show up in the deployment. Did I do something wrong? Also, I use `height ="auto"` like you suggested?

Where did you import the footer? The logo shows up on my side with this path.

**Comment by CBID2** [comment]
> > Hi @YuriDevAT! :) I revised the file path for the image import component, but for some reason, it does not show up in the deployment. Did I do something wrong? Also, I use `height ="auto"` like you suggested?
> 
> Where did you import the footer? The logo shows up on my side with this path.

I didn’t commit that line of yet. When I tried it locally, nothing showed up. For context, I did the import like this:
`import Footer from “./components/Footer”`

**Comment by CBID2** [comment]
Hi @YuriDevAT. I managed to get the footer to show. I also added the screenshots in the Before and After section. Do you mind if I add my code to the `index.js` file? I know it's not required, but it would quicken the website building process. :)

**Comment by YuriDevAT** [comment]
> Hi @YuriDevAT. I managed to get the footer to show. I also added the screenshots in the Before and After section. Do you mind if I add my code to the `index.js` file? I know it's not required, but it would quicken the website building process. :)

Ah, I see now where the miscommunication lies. My apologies 🙏🏽 . 

> - [ ] Footer showing on bottom of the page

was meant to implement it for everyone to see, therefore it is needed to import the footer component in the `index.js` file.

**Comment by YuriDevAT** [comment]
> > > Hi @YuriDevAT! :) I revised the file path for the image import component, but for some reason, it does not show up in the deployment. Did I do something wrong? Also, I use `height ="auto"` like you suggested?
> > 
> > 
> > Where did you import the footer? The logo shows up on my side with this path.
> 
> I didn’t commit that line of yet @YuriDevAT. When I tried it locally, nothing showed up. For context, I did the import like this: `import Footer from “./components/Footer”`

It's working now, right? The logo with the original path is showing up in your local environment.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:17]: Is this class in use? If not, then it can be deleted.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:16]: Concerning accessibility, think of a better alt text for the logo, or if alt text is even needed.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:13]: Isn't necessary to fulfill the AC 👇🏽 

> - [ ] No additional styling needed

**Review by YuriDevAT — CHANGES_REQUESTED** [review]
We are getting there 😄 Small adjustments, then there is only the position task left.

Hang in there 🙌🏽

File: pages/components/Footer.js
Comment (CBID2) [reply pages/components/Footer.js:13]: Noted

File: pages/components/Footer.js
Comment (CBID2) [reply pages/components/Footer.js:16]: Hi @YuriDevAT! :) I think it's best to keep the alt text so it can be read by screen readers.

**Comment by CBID2** [comment]
> > > > Hi @YuriDevAT! :) I revised the file path for the image import component, but for some reason, it does not show up in the deployment. Did I do something wrong? Also, I use `height ="auto"` like you suggested?
> > > 
> > > 
> > > Where did you import the footer? The logo shows up on my side with this path.
> > 
> > 
> > I didn’t commit that line of yet @YuriDevAT. When I tried it locally, nothing showed up. For context, I did the import like this: `import Footer from “./components/Footer”`
> 
> It's working now, right? The logo with the original path is showing up in your local environment.

Yes

File: pages/components/Footer.js
Comment (YuriDevAT) [reply pages/components/Footer.js]: You are right, alt text is for being read by screen reader users, but that does not necessarily mean you should add one. Especially when it comes to logos (which does not link to anything, yet), the alt text should be left empty like this `alt=""`.
Read this article about alt text decision tree, it is really helpful to understand the need of alt text better https://www.w3.org/WAI/tutorials/images/decision-tree/

Comment (YuriDevAT) [reply pages/components/Footer.js]: A detailed alt text is typical and very good when it comes to images, I never saw it for logos, though. Logos are mostly used as links (like, Go To Homepage, or Go To Website when it is a logo which links to another companies website). But at the current state, the logo does not link anywhere, so no alt text needed (it will be a future issue, though, so keep your eyes open 👀 )

❗ But I like how detailed you are able to describe even a logo, this is a skill not many people have. Hopefully, you make good use of it in the future, when alt text is needed on descriptive images 🙌🏽

File: pages/index.js
Comment (YuriDevAT) [inline pages/index.js:18]: Comes beneath the <main> 👍🏽

File: pages/index.js
Comment (YuriDevAT) [inline pages/index.js:6]: Please add space between imports and function for better readability and code consistency.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:13]: Not needed. Please stick to the AC. Also discussed on LinkedIn 👇🏽 

>  just go without position, just leave the width:100% and we are good to go 😊

Conversation on additional styling

**Next steps** 👉🏽 Remove everything except the width. Only do as much as described in the AC to get your PRs merged. The design of the footer will look nothing alike as it is now. So, all this code is going to be removed with the next ticket for the footer already, which is unnecessary work for the next contributor as well as for the maintainer mentioning this step in the ticket.

**Wish** 🙏🏽 : Please talk and explain reasoning behind your decisions for me to understand certain things better. If there are things you do not understand, ask me! This prevents unnecessary work for both of us which is very much appreciated to create a smooth work experience.

File: pages/components/Footer.js
Comment (YuriDevAT) [inline pages/components/Footer.js:16]: Please remove space between quotes:
Now: `alt=" "`
After: `alt=""`

**Review by YuriDevAT — CHANGES_REQUESTED** [review]
Hey you 😄 Last time revising your PR. Small adjustments but then it should be ready to go.

Please read my comments carefully and make sure you understand everything I mentioned as well as the ACs. If not, please ask me. Less code is more in this case. You will get a feeling for it soon, you will see 😎

File: pages/components/Footer.js
Comment (CBID2) [reply pages/components/Footer.js:13]: Thanks for your input with this @YuriDevAT. I was going back and forth with using Flexbox. I wanted to show of my knowledge a bit more (plus Flexbox is easier to use in my experience), but I wanted to follow the task list. Unfortunately, I went with my slightly more ambitious side. No worries, I'll definitely stick with your suggestions this time.

File: pages/index.js
Comment (CBID2) [reply pages/index.js:6]: How does that look @YuriDevAT?

Comment (YuriDevAT) [reply pages/index.js:6]:
```
import Content from "../components/content";
import Header from "./components/Header";
import Hero from "../components/hero/Hero";
import Footer from "./components/Footer";

export default function Home() {
...
}
```

File: pages/components/Footer.js
Comment (YuriDevAT) [reply pages/components/Footer.js]: I know the feeling all to well 😁. But you will be able to show off your skills when we dive deeper into the code base, I promise 🤞🏽

**Comment by CBID2** [comment]
Hi @YuriDevAT! I read your comments carefully like you suggested and made the adjustments

File: pages/index.js
Comment (CBID2) [reply pages/index.js:7]: Got it!

**Comment by YuriDevAT** [comment]
You did it 🥳 Thank you so much for your contribution, @CBID2 💟 

I am sooo grateful for this experience. This PR taught me a lot about myself, how I have to improve my communication skills, and that there are so many things I have to consider and work on to be a good maintainer. Thank you. Couldn't have done it without you 🙏🏽 

Looking forward to your next contribution, Christine 😄

**Pull request merged**
